### DISCUSSION TITLE
Why I Can't Use LLDB Which Comes From NDK

### DISCUSSION THREAD

**Discussion by SsageParuders** [android/ndk#1748] [Q&A] [answered]
### Background

> I used gdb and gdb-server from NDK to debug my native projects.
> But gdb has been removed from NDK 24.
> So I want use lldb which comes from NDK to debug my native projects.

- Os: MacOs 12.4
- NDK Version: 25.0.8775105
- LLVM Version: 14.0.6 based on r450784d

### What's I had tried

- First of all, I use lldb tools from llvm's bin directly.
   > But i get the erros,it seems that we don't have python libs
   ```zsh
    ❯ ./lldb --version
    dyld[32678]: Library not loaded: @rpath/libpython3.9.dylib
    Referenced from: /Users/hongqing/Library/Android/sdk/ndk/25.0.8775105/toolchains/llvm/prebuilt/darwin-x86_64/lib64/liblldb.14.0.6git.dylib
    Reason: tried: '/Users/hongqing/Library/Android/sdk/ndk/25.0.8775105/toolchains/llvm/prebuilt/darwin-x86_64/lib64/../lib64/libpython3.9.dylib' (no such file), '/Users/hongqing/Library/Android/sdk/ndk/25.0.8775105/toolchains/llvm/prebuilt/darwin-x86_64/bin/../lib64/libpython3.9.dylib' (no such file), '/Users/hongqing/Library/Android/sdk/ndk/25.0.8775105/toolchains/llvm/prebuilt/darwin-x86_64/bin/../lib64/libpython3.9.dylib' (no such file), '/usr/local/lib/libpython3.9.dylib' (no such file), '/usr/lib/libpython3.9.dylib' (no such file)
    [1]    32678 abort      ./lldb --version
   ```

- Then, I notice that here is a lldb.sh for us in llvm's bin.
   And I think it can help us set the right environment for lldb.
   So, I use `./lldb.sh --version` to have a test.
   Fortunately, I see the information I am looking for!
   ```zsh
    ❯ ./lldb.sh --version
    lldb version 14.0.6git (/Volumes/Android/buildbot/src/android/llvm-toolchain/out/llvm-project/lldb revision 4c603efb0cca074e9238af8b4106c30add4418f6)
    clang revision 4c603efb0cca074e9238af8b4106c30add4418f6
    llvm revision 4c603efb0cca074e9238af8b4106c30add4418f6
    ```

- So, I think we just need to use lldb.sh instead of lldb bin or `source lldb.sh` before use lldb bin.

### The Problem Now

> But when I want to enter the LLDB command line.
> Whether I use lldb.sh or lldb bin after source lldb.sh.
> It comes a information like: terminals database is inaccessible

- Just as :
  ```zsh
  ❯ ./lldb.sh
  terminals database is inaccessible
  ❯ source lldb.sh
  terminals database is inaccessible
  ❯ ./lldb
  terminals database is inaccessible
  ```

- So, how can i enter the LLDB command line?

### END

Could everyone help me? Thanks very much~

**Comment by DanAlbert** [comment, marked as answer]
Because apparently no one tested lldb on macOS. Please file a bug.

**Reply by SsageParuders** [reply]
Dear bro, you mean I need to add a bug log here?(In fact ,I had send all logs here)
Or, I need to  create an issue from this discussion?
Sorry for my bad comprehension.
Thanks for your reply~

**Reply by DanAlbert** [reply]
File a bug using https://github.com/android/ndk/issues/new/choose so we have the information asked for in the template and so we can triage it to a release. I can't triage a Q&A post.

**Reply by SsageParuders** [reply]
Thanks , will do it later.
